homeassistant/components/light/__init__.py

```diff
@@ -141,6 +146,7 @@ def preprocess_turn_on_alternatives(params):
     if profile is not None:
         params.setdefault(ATTR_XY_COLOR, profile[:2])
         params.setdefault(ATTR_BRIGHTNESS, profile[2])
+        params.setdefault(ATTR_TRANSITION, profile[3] or 0)
```

Comment:
This won't work. A list with 3 entries will throw an `IndexError` if you try to get the 4th element. 
```suggestion
        params.setdefault(ATTR_TRANSITION, profile[3] if len(profile) > 3 else 0)
```